### ISSUE TITLE
warning: 'fetch_dw_noswap' is static but used in inline function 'z900_vfetch8' which is not static

### ISSUE THREAD

**Issue by Fish-Git** [SDL-Hercules-390/hyperion#399]
___
_**NOTE:** &nbsp;issues [\#392](https://github.com/SDL-Hercules-390/hyperion/issues/392)_ "WHY static inline functions in header files?" _and [\#396](https://github.com/SDL-Hercules-390/hyperion/issues/396)_ "The most recent commit fails to build on Ubuntu" _are both closely related to this issue._
___
As mentioned by Bill Lewis in [comment 869888648](https://github.com/SDL-Hercules-390/hyperion/issues/396#issuecomment-869888648) in GitHub Issue [#396](https://github.com/SDL-Hercules-390/hyperion/issues/396), as a result of recent commits that changed many of our `static inline` functions to `extern inline` functions instead (commit 49e2723ea19d556473be372f3a14354f848bd18a followed by commit 328266663492b8ce7d7473000f6ca441750421d4, as described in GitHub Issue [#392](https://github.com/SDL-Hercules-390/hyperion/issues/392)), we are now experiencing many warnings similar to those listed below regarding static-inline functions being called by non-static (i.e. extern-inline) functions, which essentially defeats the whole purpose of using extern-inline functions in the first place! (See GitHub Issue [#392](https://github.com/SDL-Hercules-390/hyperion/issues/392) for details)

```
machdep.h:753:32: warning: ‘fetch_dw_noswap’ is static but used in inline function ‘z900_vfetch8’ which is not static
   #define fetch_dw(_p) CSWAP64(fetch_dw_noswap((_p)))
                                ^~~~~~~~~~~~~~~
[...]

../machdep.h:769:28: warning: ‘store_dw_noswap’ is static but used in inline function ‘z900_vstore8’ which is not static
   #define store_dw(_p, _v) store_dw_noswap((_p), CSWAP64((_v)))
                            ^~~~~~~~~~~~~~~
```
&nbsp;
This GitHub Issue is being created so that we don't forget to eventually fix this problem. Basically, all of our `machdep.h` static-inline functions need to be converted to extern-inline functions instead, just like what was done in the second commit mentioned above.**`(*)`**

I suspect once the above warning are eventually fixed we'll probably pick up yet another 0.5-1.0% (or more!)  improvement in performance.

&nbsp;
**`(*)`** _The first commit did things wrong (it worked on Windows using Visual C but not on Linux using gcc) whereas the second commit did things correctly so that our inline functions now build cleanly on both  platforms (except for the above mentioned warnings of course!)_

**Comment by Fish-Git** [comment]
_**FYI:**_ &nbsp; _Apparently_ the "problem" that this warning is trying to warn us about is the [likely cause of Hercules fatally crashing when my skeys patch is applied.](https://github.com/SDL-Hercules-390/hyperion/issues/400#issuecomment-872072944)

**Comment by Fish-Git** [comment]
Working on it folks!

_(Sigh... I hate this gcc crap...)_

**Comment by Fish-Git** [comment]
Fixed by commits 173bb145832e1e7cfb87cdfb354c9d8c70439cda and 38d6835b7460737a83aca078c259c11167a6ef63.

Closing.
